include/deal.II/matrix_free/operators.h

```diff
@@ -999,6 +1034,49 @@ namespace MatrixFreeOperators
       }
   }
 
+
+
+  template <int dim, int fe_degree, int n_components, typename Number>
+  inline void
+  CellwiseInverseMassMatrix<dim, fe_degree, n_components, Number>::
+    transform_from_q_points_to_basis(const unsigned int n_actual_components,
+                                     const VectorizedArray<Number> *in_array,
+                                     VectorizedArray<Number> *out_array) const
+  {
+    const unsigned int dofs_per_cell =
+      Utilities::fixed_int_power<fe_degree + 1, dim>::value;
+    internal::EvaluatorTensorProduct<internal::evaluate_evenodd,
+                                     dim,
+                                     fe_degree + 1,
+                                     fe_degree + 1,
+                                     VectorizedArray<Number>>
+      evaluator(AlignedVector<VectorizedArray<Number>>(),
+                AlignedVector<VectorizedArray<Number>>(),
+                inverse_shape);
+
+    for (unsigned int d = 0; d < n_actual_components; ++d)
+      {
+        const VectorizedArray<Number> *in  = in_array + d * dofs_per_cell;
+        VectorizedArray<Number> *      out = out_array + d * dofs_per_cell;
+
+        if (dim == 3)
+          {
+            evaluator.template hessians<2, true, false>(in, out);
```

Comment:
So you just need to apply the Hessian to change basis?

Reply:
It is not really the Hessian, it is a transformation matrix given by the inverse of the original `values` matrix. The Hessian is simply the slot I chose for the operations. This reminds me that I should update the names of these functions at some point. Let me open an issue for that.